File: inform/Project/Toolbar/IFToolbar.m
//
//  IFToolbar.m
//  Inform
//
//  Created by Toby Nelson, 2014.
//

#import "IFToolbar.h"
NSString* const IFToolbarChangedVisibility = @"IFToolbarChangedVisibility";

@implementation IFToolbar

- (void)setVisible:(BOOL)shown
{
    [super setVisible: shown];
    
	[[NSNotificationCenter defaultCenter] postNotificationName: IFToolbarChangedVisibility
														object: self];
}

@end

File: inform/Project/Toolbar/IFToolbar.swift
//
//  IFToolbar.swift
//  Inform
//
//  Created by Toby Nelson on 28/09/2023.
//

import Foundation


class IFToolbar: NSToolbar {
    @objc static let ChangedVisibility = Notification.Name("IFToolbarChangedVisibility")

    func setVisible(shown:Bool) {
        super.isVisible = shown

        NotificationCenter.default.post(name: IFToolbar.ChangedVisibility, object: self)
    }
}
